### PR TITLE
Fixes #35

### LINKED ISSUE #35: Implement format instructions
The following code:
```
let a = Decimal::new(100, 2);
let b = Decimal::new(300, 2);
let c = a / b;
println!("{}", c);
println!("{:.2}", c);
```
produces the output:
```
0.3333333333333333333333333333
0.
```

This is because the implementation converts to a string first and applies the `pad` method. Decimal should implement further format instructions to support producing 2dp without requiring rounding first up.

### PR DESCRIPTION
**Pull request by jekirl**
Adds support for formatting instructions as per #35

### CODE REVIEW

File: tests/decimal_tests.rs [hunk 1]
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
     assert_eq!("12.3456789", b.to_string());
 }
 
+// Formatting
+
+#[test]
+fn it_formats() {
+    let a = Decimal::from_str("233.323223").unwrap();
+    assert_eq!(format!("{}", a), "233.323223");
+    assert_eq!(format!("{:.9}", a), "233.323223");
```
Comment (paupino) [inline tests/decimal_tests.rs]: Shouldn't this have 9dp padded? i.e. `233.323223000`?

File: tests/decimal_tests.rs [hunk 2]
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
...
+    assert_eq!(format!("{:.0}", a), "233.");
```
Comment (paupino) [inline tests/decimal_tests.rs]: Should be `233` with no trailing `.`

File: tests/decimal_tests.rs [hunk 3]
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
...
+    assert_eq!(format!("{:.2}", a), "233.32");
+    assert_eq!(format!("{:010.2}", a), "0000233.32");
+    assert_eq!(format!("{:0<10.2}", a), "233.320000");
+}
+#[test]
+fn it_formats_neg() {
+    let a = Decimal::from_str("-233.323223").unwrap();
+    assert_eq!(format!("{}", a), "-233.323223");
+    assert_eq!(format!("{:.9}", a), "-233.323223");
```
Comment (paupino) [inline tests/decimal_tests.rs]: As above, should be `-233.323223000`

File: tests/decimal_tests.rs [hunk 4]
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
...
+    assert_eq!(format!("{:.0}", a), "-233.");
```
Comment (paupino) [inline tests/decimal_tests.rs]: As above, no trailing `.`

File: src/decimal.rs [hunk 5]
```diff
@@ -717,12 +721,7 @@ impl fmt::Display for Decimal {
             }
         }
 
-        // Last set the negation
-        if self.is_negative() {
-            rep.insert(0, '-');
-        }
-
-        f.pad(&rep)
+        f.pad_integral(self.is_positive(), "", &rep)
```
Comment (paupino) [inline src/decimal.rs]: 👍

**Comment by jekirl** [comment]
Fixed the case where `precision > scale`. I had intentionally not add padded 0s since the behavior for `precision > MAX_PRECISION` could be a bit unexpected (as perceived increase in decimal precision).  Also added more tests and a fix for the corner case of ending up with an empty string after truncating.

**Comment by paupino** [comment]
Looks good! I'm think your concern for increasing precision is warranted. I think perhaps in the future `error-chain` could be utilized and I replace some of the panic's with `Result`. The initial reason for `panic` doesn't hold true now with `?` and `map_err` etc.
